From 26723f0d2f0f9fbaf42cbb46aed6ebd1f5ffbe8b Mon Sep 17 00:00:00 2001
From: Niko Tyni <ntyni@debian.org>
Date: Fri, 8 Aug 2014 17:22:33 +0000
Subject: [PATCH 1/2] Don't answer anything to the client before it's done
 sending the POST data

LWP::Protocol::http sends long POST requests with several write() calls.
If it gets a response with a full HTTP header (up to the two newlines)
before it's done with the later write() calls sending the actual POST data,
it will stop writing. This leads to a deadlock and a test failure after
timeout.

Upstream thread at
 http://mail-archives.apache.org/mod_mbox/perl-dev/201408.mbox/%3C20140809104131.GA3670@estella.local.invalid%3E

Bug-Debian: https://bugs.debian.org/697682
---
 t/response/TestApache/read3.pm | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/t/response/TestApache/read3.pm b/t/response/TestApache/read3.pm
index 8ad9f26..2994093 100644
--- a/t/response/TestApache/read3.pm
+++ b/t/response/TestApache/read3.pm
@@ -20,8 +20,6 @@ my $expected = "foobar"x2000;
 sub handler {
     my $r = shift;
 
-    plan $r, tests => 1;
-
     # test to read data up to end of file is signaled
     my $data = '';
     my $where = 0;
@@ -31,6 +29,8 @@ sub handler {
         $where += $len;
     } while ($len > 0);
 
+    plan $r, tests => 1;
+
     ok t_cmp($data, $expected, "reading up to end of file");
 
     Apache2::Const::OK;
-- 
2.1.0.rc1

